packages/css-engine/src/core/style-sheet-regular.test.ts

```diff
@@ -21,6 +22,203 @@ const style2 = {
 const mediaRuleOptions1 = { minWidth: 300 } as const;
 const mediaId1 = "1";
 
+describe("nesting rule", () => {
+  const transformValue: TransformValue = (styleValue) => {
+    if (styleValue.type === "keyword") {
+      return {
+        type: "keyword",
+        value: styleValue.value.toUpperCase(),
+      };
+    }
+  };
+
+  test("generate rules for breakpoint", () => {
+    const sheet = createRegularStyleSheet();
+    const rule = sheet.addNestingRule(".instance");
+    rule.setDeclaration({
+      breakpoint: "base",
+      selector: "",
+      property: "width",
+      value: { type: "keyword", value: "auto" },
+    });
+    rule.setDeclaration({
```

Comment:
since you are adding multiple declarations, shouldn't this be `addDeclaration`?

Reply:
or are you making sure these are unique by property name? e.g. if I call setDeclaration with property height twice, will it add 2 declarations? if yes, then its an "add" if no, then its a set.

If there is deduplication and its a set, then maybe it would be clearer if the interface would be rule.setDeclaration('height', {rest})

Reply:
as I wrote this rule.setDeclaration('height', {rest}) I realized that this is basically setProperty

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration

In CSSOM CSSStyleDeclaration is the entire style block,can you point me to the refence for terminology you used?

Reply:
declaration here is single property but also with nested selector (state) and breakpoint. So basically same as our StyleDecl.

Reply:
are you ensuring unique declaration? if not maybe call it addDeclaration?

Reply:
Yes, ensure by unique nested selector, breakpoint and property within top level selector